styles/mixins/patterns/_forms.scss

```diff
@@ -88,3 +91,104 @@ $-color-text--disabled: var.$color-text--disabled;
     border-left: 0;
   }
 }
+
+/**
+ * Style a <label> element that wraps around its input, e.g.
+ *
+ * <label>
+ *  <input type="checkbox" />
+ *  <span>I am a label</span>
+ * </label>
+ */
+@mixin label {
+  // Apply a pointer cursor here to make sure that the label text gets a pointer
+  cursor: pointer;
+  @include layout.row($align: center);
+  @include layout.horizontal-spacing(2);
+}
+
+/**
+ * A checkbox input field.
+ *
+ * Style appearance of <input[type="checkbox"]> using an <svg> element.
+ *
+ * Style the input itself such that it is not visible to sighted users, and
+ * instead use the SVG for checkbox appearance. The SVG and input are composited
+ * such that users are still interacting with the input.
+ *
+ * See https://www.sarasoueidan.com/blog/inclusively-hiding-and-styling-checkboxes-and-radio-buttons/
+ *
+ * The <svg> element should be contained within a <span> immediately following
+ * (adjacent sibling to) the <input>:
+ *
+ *   <input class="hyp-checkbox" type="checkbox" />
+ *   <span> <!-- generated by SvgIcon -->
+ *     <svg.hyp-svg-checkbox>
+ *       <rect.hyp-svg-checkbox--background />
+ *       <polyline.hyp-svg-checkbox--checkmark />
+ *     </svg>
+ *   </span>
+ */
+@mixin checkbox {
+  // Apply a cursor here to make sure the SVG image gets a pointer
+  cursor: pointer;
+  // Style the checkbox such that it is not visible to sighted users but can
+  // be interacted with via screen-reader or touch-screen-reader.
+  position: absolute;
+  // Leaving a 1em size here allows users with touch-screen-readers to find
+  // and interact with the element.
+  width: 1em;
+  height: 1em;
+  // Some screen readers won't read out elements with 0 opacity
+  opacity: 0.00001;
+
+  // Size and align the sibling span containing the <svg>
+  & + span {
+    @include base.override {
+      // The parent container is applying horizontal-spacing.
+      // Don't add left-margin to this element.
+      // The input element, which comes first, isn't actually
+      // visible, so this (second) element should have no left-margin.
+      margin-left: var.space(0);
+    }
+    & > svg {
+      width: 1em;
+      height: 1em;
+      // Transition the checked/unchecked styling in the SVG
+      * {
+        transition: all 0.1s linear;
+      }
+    }
+  }
+
+  // Set focus styles on SVG when checkbox is focused
+  &:focus {
+    & + span > svg {
+      @include focus.outline-rule;
+    }
+
+    // TODO polyfill/browser support
```

Comment:
Both the client and lms apps load the [standard polyfill](https://www.npmjs.com/package/focus-visible) for this pseudo-class, so I think we could write code in this package to assume that either the browser natively supports `:focus-visible` or this specific polyfill is loaded by the application consuming the package.